### ISSUE TITLE
nanopolish' vcf2fasta -g

### ISSUE THREAD

**Issue by Adelaam** [jts/nanopolish#550]
Hello,

After polishing my sequence, I am trying to merged all of them and I am getting this error:

[fai_load] build FASTA index.
[fai_build_core] different line length in sequence '(null)'.
Segmentation fault (core dumped)

In the middle of polishing I have move the -g file to another folder by mistake, but I put it back later and it carried on working until it finished. Do you think was that the problem?

Thank you.

**Comment by Adelaam** [comment]
Could be because I indexed after canu and minimap?
Thank you.

**Comment by jts** [comment]
I think its because you moved the -g file in the middle of polishing. Can you try to re-run polishing?

**Comment by Adelaam** [comment]
Yes, I am polishing again and I have tried with a few of the already done vcf and I get same error. Polishing is still ongoing. Should I wait for it?

**Comment by jts** [comment]
You shouldn't see the "[fai_load] build FASTA index." line from `vcf2fasta`. I think there is something wrong with your draft genome faidx file (produced by samtools). 

Maybe try stopping your polishing, deleting the .fai files then running `samtools faidx draft_genome.fa` before starting polishing.

**Comment by Adelaam** [comment]
Ok, I am going to try it and let you know.

Thank you

**Comment by Adelaam** [comment]
! /usr/bin/python3 '/home/markwilks/nanopolish/scripts/nanopolish_makerange.py' '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/canu/52184095915.unitigs.fasta' | parallel --results nanopolish.results -P 8 \ '/home/markwilks/nanopolish/nanopolish' variants --consensus -o polished.{1}.vcf -w {1} -r '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA'  -b '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/52184095915minimap/52184095915minimpa.bam' -g '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/canu/52184095915.unitigs.fasta' -t 4 --min-candidate-frequency 0.1
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
error: faidx could not get contig length for contig tig00000002
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] build FASTA index.
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] build FASTA index.
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] build FASTA index.
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] build FASTA index.
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] build FASTA index.
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
[fai_load] failed to open FASTA index /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
This after deleting fai files files and running  
`samtools faidx draft_genome.fa
Then I added the fai files again and `again: 
 '/home/markwilks/nanopolish/nanopolish' vcf2fasta -g '/home/markwilks/polished.tig00000001:0-10602.vcf' '/home/markwilks/polished.tig00000002:0-50200.vcf' '/home/markwilks/polished.tig00000002:50000-100200.vcf' '/home/markwilks/polished.tig00000002:100000-150200.vcf' '/home/markwilks/polished.tig00000002:150000-200200.vcf' '/home/markwilks/polished.tig00000002:200000-250200.vcf' '/home/markwilks/polished.tig00000002:250000-300200.vcf' '/home/markwilks/polished.tig00000002:300000-350200.vcf' '/home/markwilks/polished.tig00000002:350000-400200.vcf' '/home/markwilks/polished.tig00000002:400000-450200.vcf' '/home/markwilks/polished.tig00000002:450000-500200.vcf' '/home/markwilks/polished.tig00000002:500000-550200.vcf' '/home/markwilks/polished.tig00000002:550000-600200.vcf' '/home/markwilks/polished.tig00000002:600000-650200.vcf' '/home/markwilks/polished.tig00000002:650000-700200.vcf' '/home/markwilks/polished.tig00000002:700000-750200.vcf' '/home/markwilks/polished.tig00000002:750000-800200.vcf' '/home/markwilks/polished.tig00000002:800000-850200.vcf' '/home/markwilks/polished.tig00000002:850000-900200.vcf' '/home/markwilks/polished.tig00000002:900000-950200.vcf' '/home/markwilks/polished.tig00000002:950000-1000200.vcf' '/home/markwilks/polished.tig00000002:1000000-1050200.vcf' '/home/markwilks/polished.tig00000002:1050000-1100200.vcf' '/home/markwilks/polished.tig00000002:1100000-1150200.vcf' '/home/markwilks/polished.tig00000002:1150000-1200200.vcf' '/home/markwilks/polished.tig00000002:1200000-1250200.vcf' '/home/markwilks/polished.tig00000002:1250000-1300200.vcf' '/home/markwilks/polished.tig00000002:1300000-1350200.vcf' '/home/markwilks/polished.tig00000002:1350000-1400200.vcf' '/home/markwilks/polished.tig00000002:1400000-1450200.vcf' '/home/markwilks/polished.tig00000002:1450000-1500200.vcf' '/home/markwilks/polished.tig00000002:1500000-1550200.vcf' '/home/markwilks/polished.tig00000002:1550000-1600200.vcf' '/home/markwilks/polished.tig00000002:1600000-1650200.vcf' '/home/markwilks/polished.tig00000002:1650000-1700200.vcf' '/home/markwilks/polished.tig00000002:1700000-1750200.vcf' '/home/markwilks/polished.tig00000002:1750000-1800200.vcf' '/home/markwilks/polished.tig00000002:1800000-1850200.vcf' '/home/markwilks/polished.tig00000002:1850000-1900200.vcf' '/home/markwilks/polished.tig00000002:1900000-1950200.vcf' '/home/markwilks/polished.tig00000002:1950000-2000200.vcf' '/home/markwilks/polished.tig00000002:2000000-2050200.vcf' '/home/markwilks/polished.tig00000002:2050000-2100200.vcf' '/home/markwilks/polished.tig00000002:2100000-2150200.vcf' '/home/markwilks/polished.tig00000002:2150000-2200200.vcf' '/home/markwilks/polished.tig00000002:2200000-2250200.vcf' '/home/markwilks/polished.tig00000002:2250000-2300200.vcf' '/home/markwilks/polished.tig00000002:2300000-2350200.vcf' '/home/markwilks/polished.tig00000002:2350000-2400200.vcf' '/home/markwilks/polished.tig00000002:2400000-2450200.vcf' '/home/markwilks/polished.tig00000002:2450000-2500200.vcf' '/home/markwilks/polished.tig00000002:2500000-2550200.vcf' '/home/markwilks/polished.tig00000002:2550000-2600200.vcf' '/home/markwilks/polished.tig00000002:2600000-2650200.vcf' '/home/markwilks/polished.tig00000002:2650000-2700200.vcf' '/home/markwilks/polished.tig00000002:2700000-2750200.vcf' '/home/markwilks/polished.tig00000002:2750000-2800200.vcf' '/home/markwilks/polished.tig00000002:2800000-2850200.vcf' '/home/markwilks/polished.tig00000002:2850000-2900200.vcf' '/home/markwilks/polished.tig00000002:2900000-2950200.vcf' '/home/markwilks/polished.tig00000002:2950000-3000200.vcf' '/home/markwilks/polished.tig00000002:3000000-3050200.vcf' '/home/markwilks/polished.tig00000002:3050000-3100200.vcf' '/home/markwilks/polished.tig00000002:3100000-3150200.vcf' '/home/markwilks/polished.tig00000002:3150000-3200200.vcf' '/home/markwilks/polished.tig00000002:3200000-3250200.vcf' '/home/markwilks/polished.tig00000002:3250000-3300200.vcf' '/home/markwilks/polished.tig00000002:3300000-3350200.vcf' '/home/markwilks/polished.tig00000002:3350000-3400200.vcf' '/home/markwilks/polished.tig00000002:3400000-3450200.vcf' '/home/markwilks/polished.tig00000002:3450000-3500200.vcf' '/home/markwilks/polished.tig00000002:3500000-3550200.vcf' '/home/markwilks/polished.tig00000002:3550000-3600200.vcf' '/home/markwilks/polished.tig00000002:3600000-3650200.vcf' '/home/markwilks/polished.tig00000002:3650000-3700200.vcf' '/home/markwilks/polished.tig00000002:3700000-3750200.vcf' '/home/markwilks/polished.tig00000002:3750000-3800200.vcf' '/home/markwilks/polished.tig00000002:3800000-3850200.vcf' '/home/markwilks/polished.tig00000002:3850000-3900200.vcf' '/home/markwilks/polished.tig00000002:3900000-3950200.vcf' '/home/markwilks/polished.tig00000002:3950000-4000200.vcf' '/home/markwilks/polished.tig00000002:4000000-4050200.vcf' '/home/markwilks/polished.tig00000002:4050000-4100200.vcf' '/home/markwilks/polished.tig00000002:4100000-4150200.vcf' '/home/markwilks/polished.tig00000002:4150000-4200200.vcf' '/home/markwilks/polished.tig00000002:4200000-4250200.vcf' '/home/markwilks/polished.tig00000002:4250000-4300200.vcf' '/home/markwilks/polished.tig00000002:4300000-4350200.vcf' '/home/markwilks/polished.tig00000002:4350000-4400200.vcf' '/home/markwilks/polished.tig00000002:4400000-4450200.vcf' '/home/markwilks/polished.tig00000002:4450000-4500200.vcf' '/home/markwilks/polished.tig00000002:4500000-4550200.vcf' '/home/markwilks/polished.tig00000002:4550000-4600200.vcf' '/home/markwilks/polished.tig00000002:4600000-4650200.vcf' '/home/markwilks/polished.tig00000002:4650000-4700200.vcf' '/home/markwilks/polished.tig00000002:4700000-4750200.vcf' '/home/markwilks/polished.tig00000002:4750000-4800200.vcf' '/home/markwilks/polished.tig00000002:4800000-4847947.vcf' '/home/markwilks/polished.tig00000003:0-9609.vcf' '/home/markwilks/polished.tig00000004:0-50200.vcf' '/home/markwilks/polished.tig00000004:50000-100200.vcf' '/home/markwilks/polished.tig00000004:100000-150200.vcf' '/home/markwilks/polished.tig00000004:150000-200200.vcf' '/home/markwilks/polished.tig00000004:200000-244647.vcf' '/home/markwilks/polished.tig00000005:0-1194.vcf' '/home/markwilks/polished.tig00000006:0-1350.vcf' '/home/markwilks/polished.tig00000007:0-1432.vcf' '/home/markwilks/polished.tig00000008:0-2954.vcf' '/home/markwilks/polished.tig00000009:0-4365.vcf' '/home/markwilks/polished.tig00000010:0-3212.vcf' '/home/markwilks/polished.tig00000011:0-2103.vcf' '/home/markwilks/polished.tig00000012:0-1572.vcf' '/home/markwilks/polished.tig00000013:0-1855.vcf' > polished_genome.fa
[fai_load] build FASTA index.
[fai_build_core] different line length in sequence '(null)'.
Segmentation fault (core dumped)

**Comment by jts** [comment]
Hi,

These errors are the problem:

```
/home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA.index.fai: No such file or directory
error: could not load the index files for input file /home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA
Please run nanopolish index on your reads (see documentation)
```

It appears your reads file (`-r`) is invalid. Can you confirm the basecalled reads are in a file called `FASTA`? Have you run nanopolish index on that file?

Jared

**Comment by Adelaam** [comment]
Hello,

Can I index the original fast5 files with a FASTA file which I've filtered as a fastq and after this passed to FASTA? Because if not maybe can be the problem?
Thank you.

Adela

**Comment by jts** [comment]
If you simply filtered the reads then it should be ok. Perhaps try changing the filename to something like `filtered.fasta` then try running nanopolish index on `filtered.fasta`

**Comment by Adelaam** [comment]
it is not working, getting the same problem: 
[fai_load] build FASTA index.
[fai_build_core] different line length in sequence '(null)'.


I've used this command for polishing, if you don't mind to have a look and see if I am missing something:

! /usr/bin/python3 '/home/markwilks/nanopolish/scripts/nanopolish_makerange.py' '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/canu/52184095915.unitigs.fasta'  | parallel --results nanopolish.results -P 8 \ '/home/markwilks/nanopolish/nanopolish' variants --consensus -o polished.{1}.vcf -w {1} -r '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/FASTA' -b '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/minimap52184095915/minimap52184095915.bam' -g '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/canu/52184095915.unitigs.fasta' -t 4 --min-candidate-frequency 0.1

Thank you!

**Comment by jts** [comment]
Did you run nanopolish index?

**Comment by Adelaam** [comment]
Yes, I did with the fasta file filtered and original fast5 files

A

**Comment by jts** [comment]
Can you rename your FASTA file to something else, then index (`filtered.fasta`)?

**Comment by Adelaam** [comment]
Ah, ok. I will
Thank you!

Adela

**Comment by Adelaam** [comment]
Hello Jared,
I have changed the name of my file to filtered.fasta but still same error. 

[fai_load] build FASTA index.

Thank you

**Comment by jts** [comment]
Can you copy and paste the commands you ran here (starting with nanopolish index) and any output that was printed to the terminal?

**Comment by Adelaam** [comment]
'/home/markwilks/nanopolish/nanopolish' index -d '/media/markwilks/Elements/20190213_1525_MN24482_FAK28377_eed2d9cb/fast5_pass' '/home/markwilks/Desktop/M.massilesis/10182489/52184144984/filtered.fasta' 
[readdb] indexing /media/markwilks/Elements/20190213_1525_MN24482_FAK28377_eed2d9cb/fast5_pass
[readdb] num reads: 493779, num reads with path to fast5: 492658


! /usr/bin/python3 '/home/markwilks/nanopolish/scripts/nanopolish_makerange.py' '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/canu/52184095915.unitigs.fasta' | parallel --results nanopolish52184095915.results -P 8 \ '/home/markwilks/nanopolish/nanopolish' variants --consensus -o polished.{1}.vcf -w {1} -r '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/filtered.fasta' -b '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/minimap52184095915/minimap52184095915.bam' -g '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/canu/52184095915.unitigs.fasta' -t 4 --min-candidate-frequency 0.1
Academic tradition requires you to cite works you base your article on.
When using programs that use GNU Parallel to process data for publication
please cite:

  O. Tange (2011): GNU Parallel - The Command-Line Power Tool,
  ;login: The USENIX Magazine, February 2011:42-47.

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

To silence this citation notice: run 'parallel --citation'.

Number of variants in span (11) would exceed max-haplotypes. Variants may be missed. Consider running with a higher value of max-haplotypes!

**Comment by jts** [comment]
Hi,

It looks like it is running. Note you're using 32 threads (`-P 8`, `-t 4`). If your computer doesn't have that many cores it may take a long time to run.

Jared

**Comment by Adelaam** [comment]
I see, I will change next time.

It is still running, but I have tried this, before it finishes in case is not working again and see same error:

'/home/markwilks/nanopolish/nanopolish' vcf2fasta -g'/home/markwilks/polished.tig00000001:0-10602.vcf' '/home/markwilks/polished.tig00000002:0-50200.vcf' '/home/markwilks/polished.tig00000002:50000-100200.vcf' '/home/markwilks/polished.tig00000002:100000-150200.vcf' '/home/markwilks/polished.tig00000002:150000-200200.vcf' '/home/markwilks/polished.tig00000002:200000-250200.vcf' '/home/markwilks/polished.tig00000002:250000-300200.vcf' '/home/markwilks/polished.tig00000002:300000-350200.vcf' '/home/markwilks/polished.tig00000002:350000-400200.vcf' '/home/markwilks/polished.tig00000002:400000-450200.vcf' '/home/markwilks/polished.tig00000002:450000-500200.vcf' '/home/markwilks/polished.tig00000002:500000-550200.vcf' '/home/markwilks/polished.tig00000002:550000-600200.vcf' '/home/markwilks/polished.tig00000002:600000-650200.vcf' '/home/markwilks/polished.tig00000002:650000-700200.vcf' '/home/markwilks/polished.tig00000002:700000-750200.vcf' '/home/markwilks/polished.tig00000002:750000-800200.vcf' '/home/markwilks/polished.tig00000002:800000-850200.vcf' '/home/markwilks/polished.tig00000002:850000-900200.vcf' '/home/markwilks/polished.tig00000002:900000-950200.vcf' '/home/markwilks/polished.tig00000002:950000-1000200.vcf' '/home/markwilks/polished.tig00000002:1000000-1050200.vcf' '/home/markwilks/polished.tig00000002:1050000-1100200.vcf' '/home/markwilks/polished.tig00000002:1100000-1150200.vcf' '/home/markwilks/polished.tig00000002:1150000-1200200.vcf' '/home/markwilks/polished.tig00000002:1200000-1250200.vcf' '/home/markwilks/polished.tig00000002:1250000-1300200.vcf' '/home/markwilks/polished.tig00000002:1300000-1350200.vcf' '/home/markwilks/polished.tig00000002:1350000-1400200.vcf' '/home/markwilks/polished.tig00000002:1400000-1450200.vcf' '/home/markwilks/polished.tig00000002:1450000-1500200.vcf' '/home/markwilks/polished.tig00000002:1500000-1550200.vcf' '/home/markwilks/polished.tig00000002:1550000-1600200.vcf' '/home/markwilks/polished.tig00000002:1600000-1650200.vcf' '/home/markwilks/polished.tig00000002:1650000-1700200.vcf' '/home/markwilks/polished.tig00000002:1700000-1750200.vcf' '/home/markwilks/polished.tig00000002:1750000-1800200.vcf' '/home/markwilks/polished.tig00000002:1800000-1850200.vcf' '/home/markwilks/polished.tig00000002:1850000-1900200.vcf' '/home/markwilks/polished.tig00000002:1900000-1950200.vcf' '/home/markwilks/polished.tig00000002:1950000-2000200.vcf' '/home/markwilks/polished.tig00000002:2000000-2050200.vcf' '/home/markwilks/polished.tig00000002:2050000-2100200.vcf' '/home/markwilks/polished.tig00000002:2100000-2150200.vcf' '/home/markwilks/polished.tig00000002:2150000-2200200.vcf' '/home/markwilks/polished.tig00000002:2200000-2250200.vcf' '/home/markwilks/polished.tig00000002:2250000-2300200.vcf' '/home/markwilks/polished.tig00000002:2300000-2350200.vcf' '/home/markwilks/polished.tig00000002:2350000-2400200.vcf' '/home/markwilks/polished.tig00000002:2400000-2450200.vcf' '/home/markwilks/polished.tig00000002:2450000-2500200.vcf' '/home/markwilks/polished.tig00000002:2500000-2550200.vcf' '/home/markwilks/polished.tig00000002:2550000-2600200.vcf' '/home/markwilks/polished.tig00000002:2600000-2650200.vcf' '/home/markwilks/polished.tig00000002:2650000-2700200.vcf' '/home/markwilks/polished.tig00000002:2700000-2750200.vcf' '/home/markwilks/polished.tig00000002:2750000-2800200.vcf' '/home/markwilks/polished.tig00000002:2800000-2850200.vcf' '/home/markwilks/polished.tig00000002:2850000-2900200.vcf' '/home/markwilks/polished.tig00000002:2900000-2950200.vcf' '/home/markwilks/polished.tig00000002:2950000-3000200.vcf' '/home/markwilks/polished.tig00000002:3000000-3050200.vcf' '/home/markwilks/polished.tig00000002:3050000-3100200.vcf' '/home/markwilks/polished.tig00000002:3100000-3150200.vcf' '/home/markwilks/polished.tig00000002:3150000-3200200.vcf' '/home/markwilks/polished.tig00000002:3200000-3250200.vcf' '/home/markwilks/polished.tig00000002:3250000-3300200.vcf' '/home/markwilks/polished.tig00000002:3300000-3350200.vcf' '/home/markwilks/polished.tig00000002:3350000-3400200.vcf' '/home/markwilks/polished.tig00000002:3400000-3450200.vcf' '/home/markwilks/polished.tig00000002:3450000-3500200.vcf' '/home/markwilks/polished.tig00000002:3500000-3550200.vcf' '/home/markwilks/polished.tig00000002:3550000-3600200.vcf' '/home/markwilks/polished.tig00000002:3600000-3650200.vcf' '/home/markwilks/polished.tig00000002:3650000-3700200.vcf' '/home/markwilks/polished.tig00000002:3700000-3750200.vcf' '/home/markwilks/polished.tig00000002:3750000-3800200.vcf' '/home/markwilks/polished.tig00000002:3800000-3850200.vcf' '/home/markwilks/polished.tig00000002:3850000-3900200.vcf' '/home/markwilks/polished.tig00000002:3900000-3950200.vcf' '/home/markwilks/polished.tig00000002:3950000-4000200.vcf' '/home/markwilks/polished.tig00000002:4000000-4050200.vcf' '/home/markwilks/polished.tig00000002:4050000-4100200.vcf' '/home/markwilks/polished.tig00000002:4100000-4150200.vcf' '/home/markwilks/polished.tig00000002:4150000-4200200.vcf' '/home/markwilks/polished.tig00000002:4200000-4250200.vcf' '/home/markwilks/polished.tig00000002:4250000-4300200.vcf' '/home/markwilks/polished.tig00000002:4300000-4350200.vcf' > genome.fasta
[fai_load] build FASTA index.
[fai_build_core] different line length in sequence '(null)'.

**Comment by jts** [comment]
You have to wait for the consensus steps to finish before you can run
vcf2fasta.

**Comment by Adelaam** [comment]
This is the error after finishing:

'/home/markwilks/nanopolish/nanopolish' vcf2fasta -g '/home/markwilks/polished.tig00000001:0-10602.vcf' '/home/markwilks/polished.tig00000002:0-50200.vcf' '/home/markwilks/polished.tig00000002:50000-100200.vcf' '/home/markwilks/polished.tig00000002:100000-150200.vcf' '/home/markwilks/polished.tig00000002:150000-200200.vcf' '/home/markwilks/polished.tig00000002:200000-250200.vcf' '/home/markwilks/polished.tig00000002:250000-300200.vcf' '/home/markwilks/polished.tig00000002:300000-350200.vcf' '/home/markwilks/polished.tig00000002:350000-400200.vcf' '/home/markwilks/polished.tig00000002:400000-450200.vcf' '/home/markwilks/polished.tig00000002:450000-500200.vcf' '/home/markwilks/polished.tig00000002:500000-550200.vcf' '/home/markwilks/polished.tig00000002:550000-600200.vcf' '/home/markwilks/polished.tig00000002:600000-650200.vcf' '/home/markwilks/polished.tig00000002:650000-700200.vcf' '/home/markwilks/polished.tig00000002:700000-750200.vcf' '/home/markwilks/polished.tig00000002:750000-800200.vcf' '/home/markwilks/polished.tig00000002:800000-850200.vcf' '/home/markwilks/polished.tig00000002:850000-900200.vcf' '/home/markwilks/polished.tig00000002:900000-950200.vcf' '/home/markwilks/polished.tig00000002:950000-1000200.vcf' '/home/markwilks/polished.tig00000002:1000000-1050200.vcf' '/home/markwilks/polished.tig00000002:1050000-1100200.vcf' '/home/markwilks/polished.tig00000002:1100000-1150200.vcf' '/home/markwilks/polished.tig00000002:1150000-1200200.vcf' '/home/markwilks/polished.tig00000002:1200000-1250200.vcf' '/home/markwilks/polished.tig00000002:1250000-1300200.vcf' '/home/markwilks/polished.tig00000002:1300000-1350200.vcf' '/home/markwilks/polished.tig00000002:1350000-1400200.vcf' '/home/markwilks/polished.tig00000002:1400000-1450200.vcf' '/home/markwilks/polished.tig00000002:1450000-1500200.vcf' '/home/markwilks/polished.tig00000002:1500000-1550200.vcf' '/home/markwilks/polished.tig00000002:1550000-1600200.vcf' '/home/markwilks/polished.tig00000002:1600000-1650200.vcf' '/home/markwilks/polished.tig00000002:1650000-1700200.vcf' '/home/markwilks/polished.tig00000002:1700000-1750200.vcf' '/home/markwilks/polished.tig00000002:1750000-1800200.vcf' '/home/markwilks/polished.tig00000002:1800000-1850200.vcf' '/home/markwilks/polished.tig00000002:1850000-1900200.vcf' '/home/markwilks/polished.tig00000002:1900000-1950200.vcf' '/home/markwilks/polished.tig00000002:1950000-2000200.vcf' '/home/markwilks/polished.tig00000002:2000000-2050200.vcf' '/home/markwilks/polished.tig00000002:2050000-2100200.vcf' '/home/markwilks/polished.tig00000002:2100000-2150200.vcf' '/home/markwilks/polished.tig00000002:2150000-2200200.vcf' '/home/markwilks/polished.tig00000002:2200000-2250200.vcf' '/home/markwilks/polished.tig00000002:2250000-2300200.vcf' '/home/markwilks/polished.tig00000002:2300000-2350200.vcf' '/home/markwilks/polished.tig00000002:2350000-2400200.vcf' '/home/markwilks/polished.tig00000002:2400000-2450200.vcf' '/home/markwilks/polished.tig00000002:2450000-2500200.vcf' '/home/markwilks/polished.tig00000002:2500000-2550200.vcf' '/home/markwilks/polished.tig00000002:2550000-2600200.vcf' '/home/markwilks/polished.tig00000002:2600000-2650200.vcf' '/home/markwilks/polished.tig00000002:2650000-2700200.vcf' '/home/markwilks/polished.tig00000002:2700000-2750200.vcf' '/home/markwilks/polished.tig00000002:2750000-2800200.vcf' '/home/markwilks/polished.tig00000002:2800000-2850200.vcf' '/home/markwilks/polished.tig00000002:2850000-2900200.vcf' '/home/markwilks/polished.tig00000002:2900000-2950200.vcf' '/home/markwilks/polished.tig00000002:2950000-3000200.vcf' '/home/markwilks/polished.tig00000002:3000000-3050200.vcf' '/home/markwilks/polished.tig00000002:3050000-3100200.vcf' '/home/markwilks/polished.tig00000002:3100000-3150200.vcf' '/home/markwilks/polished.tig00000002:3150000-3200200.vcf' '/home/markwilks/polished.tig00000002:3200000-3250200.vcf' '/home/markwilks/polished.tig00000002:3250000-3300200.vcf' '/home/markwilks/polished.tig00000002:3300000-3350200.vcf' '/home/markwilks/polished.tig00000002:3350000-3400200.vcf' '/home/markwilks/polished.tig00000002:3400000-3450200.vcf' '/home/markwilks/polished.tig00000002:3450000-3500200.vcf' '/home/markwilks/polished.tig00000002:3500000-3550200.vcf' '/home/markwilks/polished.tig00000002:3550000-3600200.vcf' '/home/markwilks/polished.tig00000002:3600000-3650200.vcf' '/home/markwilks/polished.tig00000002:3650000-3700200.vcf' '/home/markwilks/polished.tig00000002:3700000-3750200.vcf' '/home/markwilks/polished.tig00000002:3750000-3800200.vcf' '/home/markwilks/polished.tig00000002:3800000-3850200.vcf' '/home/markwilks/polished.tig00000002:3850000-3900200.vcf' '/home/markwilks/polished.tig00000002:3900000-3950200.vcf' '/home/markwilks/polished.tig00000002:3950000-4000200.vcf' '/home/markwilks/polished.tig00000002:4000000-4050200.vcf' '/home/markwilks/polished.tig00000002:4050000-4100200.vcf' '/home/markwilks/polished.tig00000002:4100000-4150200.vcf' '/home/markwilks/polished.tig00000002:4150000-4200200.vcf' '/home/markwilks/polished.tig00000002:4200000-4250200.vcf' '/home/markwilks/polished.tig00000002:4250000-4300200.vcf' '/home/markwilks/polished.tig00000002:4300000-4350200.vcf' '/home/markwilks/polished.tig00000002:4350000-4400200.vcf' '/home/markwilks/polished.tig00000002:4400000-4450200.vcf' '/home/markwilks/polished.tig00000002:4450000-4500200.vcf' '/home/markwilks/polished.tig00000002:4500000-4550200.vcf' '/home/markwilks/polished.tig00000002:4550000-4600200.vcf' '/home/markwilks/polished.tig00000002:4600000-4650200.vcf' '/home/markwilks/polished.tig00000002:4650000-4700200.vcf' '/home/markwilks/polished.tig00000002:4700000-4750200.vcf' '/home/markwilks/polished.tig00000002:4750000-4800200.vcf' '/home/markwilks/polished.tig00000002:4800000-4847947.vcf' '/home/markwilks/polished.tig00000003:0-9609.vcf' '/home/markwilks/polished.tig00000004:0-50200.vcf' '/home/markwilks/polished.tig00000004:50000-100200.vcf' '/home/markwilks/polished.tig00000004:100000-150200.vcf' '/home/markwilks/polished.tig00000004:150000-200200.vcf' '/home/markwilks/polished.tig00000004:200000-244647.vcf' '/home/markwilks/polished.tig00000005:0-1194.vcf' '/home/markwilks/polished.tig00000006:0-1350.vcf' '/home/markwilks/polished.tig00000007:0-1432.vcf' '/home/markwilks/polished.tig00000008:0-2954.vcf' '/home/markwilks/polished.tig00000009:0-4365.vcf' '/home/markwilks/polished.tig00000010:0-3212.vcf' '/home/markwilks/polished.tig00000011:0-2103.vcf' '/home/markwilks/polished.tig00000012:0-1572.vcf' '/home/markwilks/polished.tig00000013:0-1855.vcf' > 52184095915polished.fasta
[fai_load] build FASTA index.
[fai_build_core] different line length in sequence '(null)'.
Segmentation fault (core dumped)

**Comment by jts** [comment]
Are any of those VCF files empty?

**Comment by Adelaam** [comment]
Yes, there are some empty VCF files. Should I delete them? 
Why does it happen?

Thank you.

Adela

**Comment by jts** [comment]
It probably means some of the tasks ran out of memory because you tried to run too many at once. Try changing the `-P` parameter to `parallel` to `-P 1`.

**Comment by Adelaam** [comment]
Hello,

I have tried with -p 1 and I got the same results one of them is empty. Is contig number 3 so I have tried to polished with variants --consensus alone and I got same result, an empty file and the next message:
Warning: 9194 variants in span, region not called [5895 9600]
[post-run summary] total reads: 6, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 1, bad fast5: 0

Thank you

Adela

**Comment by Adelaam** [comment]
Do you think should I deleted that one?

**Comment by Adelaam** [comment]
Hello,

Still having this problem, do you know how can I fix it?
Thank you

**Comment by jts** [comment]
Hi @Adelaam,

Sorry for not replying. If you see the "post-run" summary message it means the process ran correctly so you should have a valid VCF output file. Did you see any other messages? 

The "variants in span" warning is worrying, it might suggest there is a problem with your assembly.

Jared

**Comment by Adelaam** [comment]
Hi Jared,

Thank you for your answer.

No, no any other message. The VCF output file I got, has all the tittles but nothing else.
Do you know what kind of problem could be?

Adela

**Comment by jts** [comment]
Oh, if it has the header information in the VCF it should be ok. Do you still get an error from vcf2fasta?

**Comment by Adelaam** [comment]
Yes, same:

'/home/markwilks/nanopolish/nanopolish' vcf2fasta -g '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/nanopolish52184095915.results/polished.tig00000001:0-10602.vcf'.....................> '/home/markwilks/Desktop/M.massilesis/10182489/52184095915/nanopolish52184095915.results/polished52184095915.fa'
[fai_load] build FASTA index.
[fai_build_core] different line length in sequence '(null)'.
Segmentation fault (core dumped)

**Comment by jts** [comment]
Can you send me all of the vcf files, and the genome?

**Comment by Adelaam** [comment]
Yes, I can.

Do you want the fasta file for genome or the one after canu?
Let me know how to send all this to you.
A

**Comment by jts** [comment]
I want the file that you're passing as the `-g` parameter to `vcf2fasta`. If the files are too large to email or attach to this issue then maybe host it on dropbox or google drive?

**Comment by Adelaam** [comment]
Have you got it?
A

**Comment by jts** [comment]
No I didn't, what email did you send it to ?

**Comment by Adelaam** [comment]
To jts/nanopolish

**Comment by jts** [comment]
You have to attach the file using the github UI, you can't reply to the issue thread in email

**Comment by Adelaam** [comment]
[folder.zip](https://github.com/jts/nanopolish/files/2987559/folder.zip)

**Comment by jts** [comment]
Thanks. I will try to look at this soon.

**Comment by jts** [comment]
I don't see the see the genome file, can you include that as well?

**Comment by Adelaam** [comment]
What do you mean the one after canu or the one after filtering?

**Comment by jts** [comment]
I just looked over your command again and noticed a problem. The correct command should be:

```
nanopolish vcf2fasta -g draft_genome.fa a.vcf b.vcf ...
```

where `draft_genome.fa` is the genome that you polished with `nanopolish variants`. It looks like you passed a vcf file to the `-g` argument, which might have caused this error.

**Comment by Adelaam** [comment]
[genome.zip](https://github.com/jts/nanopolish/files/2988400/genome.zip)

**Comment by jts** [comment]
This works for me.

```
>nanopolish vcf2fasta -g 52184095915.unitigs.fasta *.vcf > polished.fasta
[vcf2fasta] rewrote contig tig00000001 with 4 subs, 17 ins, 0 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000002 with 307 subs, 13519 ins, 96 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000003 with 0 subs, 0 ins, 0 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000004 with 31 subs, 868 ins, 6 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000005 with 3 subs, 6 ins, 0 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000006 with 2 subs, 6 ins, 0 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000007 with 3 subs, 8 ins, 2 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000008 with 3 subs, 16 ins, 1 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000009 with 15 subs, 24 ins, 3 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000010 with 9 subs, 16 ins, 2 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000011 with 10 subs, 15 ins, 3 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000012 with 5 subs, 4 ins, 0 dels (0 skipped)
[vcf2fasta] rewrote contig tig00000013 with 7 subs, 10 ins, 1 dels (0 skipped)
```

**Comment by Adelaam** [comment]
I got it! 
Thank you very much!!
Adela

**Comment by jts** [comment]
Glad to hear it, sorry for the difficulty.

**Comment by Adelaam** [comment]
It is ok.

Thank you for your support with it.


________________________________
